File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/DanhMucTinTucColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DanhMucTinTucRow } from "./DanhMucTinTucRow";

export interface DanhMucTinTucColumns {
    MaDm: Column<DanhMucTinTucRow>;
    TenDm: Column<DanhMucTinTucRow>;
    Slug: Column<DanhMucTinTucRow>;
    MoTa: Column<DanhMucTinTucRow>;
    ThuTu: Column<DanhMucTinTucRow>;
    HienThi: Column<DanhMucTinTucRow>;
}

export class DanhMucTinTucColumns extends ColumnsBase<DanhMucTinTucRow> {
    static readonly columnsKey = 'SanPhamDB.DanhMucTinTuc';
    static readonly Fields = fieldsProxy<DanhMucTinTucColumns>();
}

File: wheystore/wheystore.Web/Modules/SanPhamDB/DanhMucTinTuc/DanhMucTinTucColumns.cs
namespace wheystore.SanPhamDB.Columns;

[ColumnsScript("SanPhamDB.DanhMucTinTuc")]
[BasedOnRow(typeof(DanhMucTinTucRow), CheckNames = true)]
public class DanhMucTinTucColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int MaDm { get; set; }
    [EditLink]
    public string TenDm { get; set; }
    public string Slug { get; set; }
    public string MoTa { get; set; }
    public int ThuTu { get; set; }
    public bool HienThi { get; set; }
}